### ISSUE TITLE
undoStopBefore not placed before deleting

### ISSUE THREAD

**Issue by superle3** [Oskar-Idland/hsnips#3]
continuing from https://github.com/draivin/hsnips/issues/189 where vscode broke their api after 1.85.2.

test snippet
```ts
snippet trigger1 "trigger1" iA
hello world
endsnippet
```

expected behaviour:
`trigger1` -> `hello world` C-Z -> `trigger1`

current behaviour:
`trigger1` -> `hello world` C-Z -> ` `
where C-Z is the undo shortcut.

it probably has to do with the code below in extension.ts, where there is no undostop placed when the snippet is deleted. 
```ts
await editor.edit(
        (eb) => {
            eb.delete(snippetExpansion ? completion.completionRange : completion.range);
        },
        { undoStopAfter: false, undoStopBefore: !snippetExpansion }
    );
```

**Comment by Oskar-Idland** [comment]
Yes, I have noticed, but always just accepted that you have to press twice to get what was there before the snippet. It seems like OrangeX4 had encountered this, but it was made like this to avoid some bug with VIM mode.

**Comment by Oskar-Idland** [comment]
Just by quick reading and tweaking the parameters I cold not make much sense of it. It made no difference if `undoStopAFter` was true or false, but the `undoStopBefore` had to be `!snippetExpansion`.

**Comment by kylemonette** [comment]
Is there a way to make it so that C-Z (undo shortcut) does not have to be pressed twice?

**Comment by superle3** [comment]
with the most recent version, no. Vscode's extension api is kinda broken/badly documented. We have tried to get a fix, but got shut down without any reason so there is no way to fix it now and probably also not in the future. 

using the old plugin of dravin with vsc version 1.85.2 does work with the undo, as that is the latest version before they broke their api.
